### PR TITLE
Various clean up and fixes

### PR DESCRIPTION
**Pull request by kubeliv**
Made some various changes to cleanup getEvents(dateRange), and other beneficial changes, along with ones that were done and merged in already, I'll need to remove these before merging.

EDIT 1:
Made change that will fix the createEvent action from being created. This happened on Google Calendars.

EDIT 2:
Work is in progress for support for guests attached to an event.

### CODE REVIEW

File: src/main/java/org/dsa/iot/calendar/abstractions/BaseCalendar.java [hunk 1]
```diff
@@ -32,7 +32,18 @@ public BaseCalendar(Node eventsNode) {
      * @param end End time
      * @return List of strings to unique ids.
      */
-    public abstract List<DSAEvent> getEvents(Date start, Date end);
+    public List<DSAEvent> getEvents(Date start, Date end) {
+        List<DSAEvent> events = getEvents();
+        List<DSAEvent> newEvents = new ArrayList<>();
+
+        for (DSAEvent event : events) {
+            if (event.checkInRange(start, end)) {
```
Comment (Pacane) [inline src/main/java/org/dsa/iot/calendar/abstractions/BaseCalendar.java]: I'd rename this method to `isInRange` or `isBetween`

File: src/main/java/org/dsa/iot/calendar/abstractions/BaseCalendar.java [hunk 2]
```diff
@@ -32,7 +32,18 @@ public BaseCalendar(Node eventsNode) {
      * @param end End time
      * @return List of strings to unique ids.
      */
-    public abstract List<DSAEvent> getEvents(Date start, Date end);
+    public List<DSAEvent> getEvents(Date start, Date end) {
+        List<DSAEvent> events = getEvents();
+        List<DSAEvent> newEvents = new ArrayList<>();
```
Comment (Pacane) [inline src/main/java/org/dsa/iot/calendar/abstractions/BaseCalendar.java:37]: I'd rename this `eventsInRange`.

File: src/main/java/org/dsa/iot/calendar/google/GoogleCalendar.java [hunk 3]
```diff
@@ -47,8 +47,8 @@ public GoogleCalendar(Node calendarNode, String clientId, String clientSecret) t
         super(calendarNode.getChild("events"));
         this.clientId = clientId;
         this.clientSecret = clientSecret;
-        this.httpTransport = new NetHttpTransport();
-        this.jsonGenerator = JacksonFactory.getDefaultInstance();
+        httpTransport = new NetHttpTransport();
+        jsonGenerator = JacksonFactory.getDefaultInstance();
```
Comment (Pacane) [inline src/main/java/org/dsa/iot/calendar/google/GoogleCalendar.java:49]: It'd be great if we used dependency injection where possible. Have you ever used Dagger or Guice?

**Review by Pacane — CHANGES_REQUESTED** [review]

File: src/main/java/org/dsa/iot/calendar/google/GoogleCalendar.java (hunk 3 above)
Comment (kubeliv) [reply src/main/java/org/dsa/iot/calendar/google/GoogleCalendar.java:49]: @Pacane Is this something that we can visit in a later PR?

Comment (Pacane) [reply src/main/java/org/dsa/iot/calendar/google/GoogleCalendar.java:49]: Yes

File: src/main/java/org/dsa/iot/calendar/abstractions/BaseCalendar.java (hunk 1 above)
Comment (Pacane) [reply src/main/java/org/dsa/iot/calendar/abstractions/BaseCalendar.java]: I meant the `event.checkInRange` should be renamed to `event.isInRange`

**Review by Pacane — APPROVED** [review]

File: src/main/java/org/dsa/iot/calendar/google/GoogleCalendar.java [hunk 4]
```diff
@@ -223,7 +234,7 @@ public boolean supportsMultipleCalendars() {
             for (CalendarListEntry listEntry : calendarList.getItems()) {
                 calendars.add(new DSAIdentifier(listEntry.getId(), listEntry.getSummary()));
             }
-        } catch (IOException e) {
+        } catch (Exception e) {
```
Comment (azenla) [inline src/main/java/org/dsa/iot/calendar/google/GoogleCalendar.java]: Generally you want to catch `Throwable` rather than `Exception`, just in case.

Comment (Pacane) [reply src/main/java/org/dsa/iot/calendar/google/GoogleCalendar.java]: Never seen that pattern though.

**Pull request merged**